common/src/main/java/org/cbioportal/cmo/pipelines/common/util/InstantStringUtil.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) 2023 Memorial Sloan Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal CMO-Pipelines.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.cmo.pipelines.common.util;
+
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAccessor;
+import java.time.ZoneId;
+import java.time.zone.ZoneRulesException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+import org.apache.log4j.Logger;
+
+/* parses and converts strings to required format for java.time.Instant.parse(), which requires java.time.format.DateTimeFormatter.ISO_INSTANT compatible format.
+ * Several formats are attempted for parsing, based on easily output values from the unix 'date' program (run with LC_TIME="en_US.UTF-8"):
```

Comment:
Yes. The linux 'date' command can generate strings in number of different formats. If no format specifier is passed to 'date' the output format seems to be dependent on the locale settings configured on the system. This code could have been written to require one particular format (such as iso-8601 which has the advantage of not containing spaces) but users may not be familiar with reading/writing dates in that format. This code attempts to parse the date in several formats - those which were easily available in the java time library and which were easy to convert to from a "to the second" standard output format from 'date'. Should the default format of the date command change, or if a system migration changes the locale so that it is no longer compatible, having additional recognized formats allows easy adjustment without requiring java code changes.